### PR TITLE
Allow reordering of ToDoExplorer header columns

### LINKED ISSUE #2947: RD>Tools>Todo Items>Todo Explorer Reorder header columns
Allow reordering so that it reads "Description | Module Name | Project Name | Line Number" or whatever order one would prefer.

### PR DESCRIPTION
**Pull request by IvenBach**
Close #2947. Column headers are order-able and persisted to config file. If config file doesn't have `ColumnHeaderInformation` element the default defined element is used. Column widths are saved as `Auto` and I don't forsee this being an issue. If widths need to be saved this can be revisited.

### CODE REVIEW

File: Rubberduck.Core/Properties/Settings.settings [hunk 1]
```diff
@@ -185,5 +185,26 @@
...
+    &lt;Width /&gt;
+  &lt;/DescriptionColumn&gt;
+  &lt;ProjectColumn&gt;
+    &lt;DisplayIndex&gt;2&lt;/DisplayIndex&gt;
+    &lt;Width /&gt;
+  &lt;/ProjectColumn&gt;
+  &lt;ModuleColumn&gt;
+    &lt;DisplayIndex&gt;1&lt;/DisplayIndex&gt;
+    &lt;Width /&gt;
+  &lt;/ModuleColumn&gt;
```
Comment (Vogel612) [inline Rubberduck.Core/Properties/Settings.settings]: could we have the defaults ordered by their display index?

File: Rubberduck.Core/Settings/ToDoListSettings.cs [hunk 2]
```diff
@@ -26,21 +28,83 @@ public ToDoMarker[] ToDoMarkers
...
+                && ColumnHeaderInformation.Equals(other.ColumnHeaderInformation);
+        }
+    }
+
+    public class ToDoExplorerColumns : IEquatable<ToDoExplorerColumns>
+    {
+        public ToDoExplorerColumn DescriptionColumn { get; set; }
+        public ToDoExplorerColumn ProjectColumn { get; set; }
+        public ToDoExplorerColumn ModuleColumn { get; set; }
+        public ToDoExplorerColumn LineNumberColumn { get; set; }
```
Comment (Vogel612) [inline Rubberduck.Core/Settings/ToDoListSettings.cs]: so if we wanted to make this more extensible and less prone to breakage, these could be changed into a Collection and update `ToDoExplorerColumn` itself to contain a `HeaderName` or something like that.

Comment (IvenBach) [reply Rubberduck.Core/Settings/ToDoListSettings.cs]: As in `ObservableCollection<ToDoExplorerColumn> ColumnHeadersInformation`? That's what comes to mind when you suggest a collection and then implementing `INotifyPropertyChanged` when a column`s DisplayIndex is changed.

**Comment by retailcoder** [comment]
I would honestly merge this PR as-is, I'm just concerned with the UX of having one out of (many) grids capable of persisting column positions. How do we feel about this @rubberduck-vba/dev?

**Comment by Vogel612** [comment]
@retailcoder that's a valid concern, but I'd say pulling this into working for all grids in RD is beyond the scope of a single PR. I'm not even sure how many users would notice or use this kind of feature...

**Pull request merged**